src/reconstruction/fastGapFill/AuxillaryFiles/createXMatrix2.m

```diff
@@ -224,5 +224,6 @@
     end
     showprogress(i/length(compounds));
 end
-
-ExchangeRxnMatrix.mets = ExchangeRxnMatrix.mets';
+%Currently we are likely to have a largely blown up S matrix.
+ExchangeRxnMatrix.S = ExchangeRxnMatrix.S(1:numel(ExchangeRxnMatrix.mets),1:numel(ExchangeRxnMatrix.rxns));

```

Comment:
Because the ExchangeMatrix S is initialized as a VERY large matrix. 
If you look at lines 55ff, this is a space allocation, and  actually initializes the size of the matrix.
This leads to ExchangeRxnMatrix (which is created as a COBRA model struct having a size that is inconsistent with the actual number of reactions/metabolites).